pkg/common/common_test.go

```diff
@@ -473,3 +476,163 @@ func TestGetVolumeMode(t *testing.T) {
 		}
 	}
 }
+
+func TestNodeExists(t *testing.T) {
+	node := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-node",
+		},
+	}
+
+	tests := []struct {
+		nodeAdded *v1.Node
+		// Required.
+		nodeQueried    *v1.Node
+		expectedResult bool
+	}{
+		{
+			nodeAdded:      node,
+			nodeQueried:    node,
+			expectedResult: true,
+		},
+		{
+			nodeQueried:    node,
+			expectedResult: false,
+		},
+	}
+
+	for _, test := range tests {
+		client := fake.NewSimpleClientset()
+		informers := informers.NewSharedInformerFactory(client, time.Duration(0))
+		nodeInformer := informers.Core().V1().Nodes()
+
+		if test.nodeAdded != nil {
+			nodeInformer.Informer().GetStore().Add(test.nodeAdded)
+		}
+
+		exists, err := NodeExists(nodeInformer.Lister(), test.nodeQueried.Name)
+		if err != nil {
+			t.Errorf("Got unexpected error: %s", err.Error())
+		}
+		if exists != test.expectedResult {
+			t.Errorf("expected result: %t, actual: %t", test.expectedResult, exists)
+		}
+	}
+}
+
+func TestNodeAttachedToLocalPV(t *testing.T) {
+	nodeName := "testNodeName"
+
+	tests := []struct {
+		name             string
+		pv               *v1.PersistentVolume
+		expectedNodeName string
+		expectedStatus   bool
+	}{
+		{
+			name:             "NodeAffinity will all necessary fields",
+			pv:               withNodeAffinity(pv(), nodeName, NodeLabelKey),
+			expectedNodeName: nodeName,
+			expectedStatus:   true,
+		},
+		{
+			name:             "empty Values array",
+			pv:               removeAllNodeNames(withNodeAffinity(pv(), nodeName, NodeLabelKey)),
+			expectedNodeName: "",
+			expectedStatus:   false,
+		},
+		{
+			name:             "wrong node label key",
+			pv:               withNodeAffinity(pv(), nodeName, "wrongLabel"),
+			expectedNodeName: "",
+			expectedStatus:   false,
+		},
+	}
+
+	for _, test := range tests {
+		nodeName, ok := NodeAttachedToLocalPV(test.pv)
+		if nodeName != test.expectedNodeName || ok != test.expectedStatus {
```

Comment:
I think it's better to separate checking error and comparing result into two steps. The structure will be something like:

if gotExpected := gotExpectedError(test.name, test.expectErr, err); gotExpected != nil {
			t.Fatal(gotExpected)
		}
if nodeName != test.expectedNodeName {
t.Errorf("got XXX expected XXX...")
}